akka-actor-typed/src/main/scala/akka/actor/typed/internal/pubsub/TopicImpl.scala

```diff
@@ -77,8 +77,15 @@ private[akka] final class TopicImpl[T](topicName: String, context: ActorContext[
 
     case Publish(message) =>
       if (topicInstances.isEmpty) {
-        context.log.trace("Publishing message of type [{}] but no subscribers, dropping", msg.getClass)
-        context.system.deadLetters ! Dropped(message, "No topic subscribers known", context.self.toClassic)
+        if (localSubscribers.isEmpty) {
+          context.log.trace("Publishing message of type [{}] but no subscribers, dropping", msg.getClass)
+          context.system.deadLetters ! Dropped(message, "No topic subscribers known", context.self.toClassic)
+        } else {
+          context.log.trace(
+            "Publishing message of type [{}] to local subscribers only (topic listing not seen yet)",
+            msg.getClass)
+          localSubscribers.foreach(_ ! message)
```

Comment:
ok, so do I understand it correctly that also the local subscribers will short hereafter be added as real topicInstances?

Reply:
Yep, this problem is only when the topic actor changes from no local subscribers to one subscriber, since the topic defers registering itself with the receptionist until it has at least one subscriber (and also de-registers itself from the receptionist when the last local subscriber unsubscribes) there is a short time period where you could ask it for stats, see that there is a (local) subscriber, but the topic has not yet gotten the update from the receptionist.

When I think about it, there probably is _another_ trickier, but maybe not as bad, race condition when the last local subscriber unsubscribes, the topic deregisters itself from the service key, and gets that update, but there are other topic instances known to the receptionist. 

When a new actor subscribes in that scenario, there will be a short time period when a published message will go to all the other topics, before the receptionist update arrives to the topic actor, even though the local subscriber is known before that.

Not entirely sure it is something we need to do anything about though, the scenario fixed here where the message was dropped even though a local subscriber is known seems worse.

Reply:
I agree, that seems less important